downloads/swan-lake-release-notes/swan-lake-preview5/RELEASE_NOTE.md

```diff
@@ -0,0 +1,255 @@
+
+This release is the fifth preview version of Ballerina Swan Lake. This release includes a new set of language features along with improvements and bug fixes to the compiler, runtime, standard libraries, and developer tooling.
+
+- [Updating Ballerina](#updating-ballerina)
+    - [For existing users](#for-existing-users)
+    - [For new users](#for-new-users)
+- [What is new in Ballerina Swan Lake Preview 5](#what-is-new-in-ballerina-swan-lake-preview-5)
+    - [Language](#language)
+        - [Isolated objects](#isolated-objects)
+        - [Removed XML functional constructors](#removed-xml-functional-constructors)
+        - [Unified object method scope and object field scope](#unified-object-method-scope-and-object-field-scope)
+        - [Improved lang library functions](#improved-lang-library-functions)
+    - [Runtime](#runtime)
+    - [Standard Library](#standard-library)
+        - [HTTP client remote function API improvement with data-binding support](#http-client-remote-function-api-improvement-with-data-binding-support)
+        - [SQL procedure call API improvement](#sql-procedure-call-api-improvement)
+        - [Add misfire policy](#add-misfire-policy)
+    - [Developer Tools](#developer-tools)
+        - [Code formatting](#code-formatting)
+
+### Updating Ballerina
+
+You can use the update tool to update to Ballerina Swan Lake Preview 5 as follows.
+
+#### For existing users
+
+If you are already using Ballerina, you can directly update your distribution to the Swan Lake channel using the [Ballerina Update Tool](http://ballerina.io/swan-lake/learn/keeping-ballerina-up-to-date/). To do this, first, execute the command below to get the update tool updated to its latest version. 
+                        
+> `ballerina update`
+
+ Next, execute the command below to update to Swan Lake Preview 5.
+
+ > `ballerina dist pull slp5b1`                 
+
+However, if you are using a Ballerina version below 1.1.0, install via the [installers](https://ballerina.io/downloads/).
+
+#### For new users
+
+If you have not installed Ballerina, then download the [installers](https://ballerina.io/downloads/) to install.
+
+### What is new in Ballerina Swan Lake Preview 5
+
+#### Language
+
+##### Isolated objects
+
+A class definition or an object-type descriptor can now be marked as `isolated`. All fields of an `isolated` object that are not `final` or are not a subtype of `readonly` or `isolated object {}` have to be `private` fields.
+
+The methods of an `isolated` object that access such fields can only refer to `self` within a `lock` statement, which ensures that mutable state is only accessed within `lock` statements. If a method accesses such a field, additional rules apply to how values are transferred in and out in order to ensure that there are no references into the object’s mutable storage except via isolated objects. The initial value expressions of the fields also need to ensure the same.
```

Comment:
The first sentence, doesn't it feel like the info in the second part of the sentence is actually a repetition of what's in the first part? Shouldn't the second part be explaining why such a restriction is in place instead?

Reply:
@MaryamZi

Reply:
Do you mean the part on transferring in/out? That's actually independent of the first part.

Reply:
No I meant the following sentence: "The methods of an `isolated` object that access such fields can only refer to `self` within a `lock` statement, which ensures that mutable state is only accessed within `lock` statements"

Reply:
Just wanted to emphasize the bit about such fields being mutable. Any suggestions?

Reply:
Yeah, I'm ok with emphasizing that. But my point was, it doesn't convey what problem it solves by restricting the use of `self` to just within `lock` statements. For example, something like "which prevents mutable state from being concurrently modified", or something of the sort, which conveys the problem this restriction addresses.

Reply:
+1, the spec actually mentions data race, how about

```suggestion
The methods of an `isolated` object that access such fields can only refer to `self` within a `lock` statement, which ensures that there is no data race in accessing mutable state. If a method accesses such a field, additional rules apply to how values are transferred in and out in order to ensure that there are no references into the object’s mutable storage except via isolated objects. The initial value expressions of the fields also need to ensure the same.
```